examples/nextjs-13/app/layout.js

```diff
@@ -0,0 +1,29 @@
+'use client';
```

Comment:
We shouldn't move forward with this PR before we are able to get rid of this directive. The layout shouldn't be a client component.

Reply:
> It feels like it was created in a bubble, without considering how real-life app constraints are. For a given request, a theme is static, a locale is static, etc.
Exactly!

Reply:
> We shouldn't move forward with this PR before we are able to get rid of this directive. The layout shouldn't be a client component.

Per https://github.com/mui/material-ui/issues/34905#issuecomment-1401306594, I think that it would be great to merge to have this demo. The /app folder has more to offer beyond server-side components.